presto-orc/src/test/java/com/facebook/presto/orc/BenchmarkMapFlatWriter.java

```diff
@@ -0,0 +1,202 @@
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+
+import static com.facebook.presto.common.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.metadata.FunctionAndTypeManager.createTestFunctionAndTypeManager;
+import static com.facebook.presto.orc.DwrfEncryptionProvider.NO_ENCRYPTION;
+import static com.facebook.presto.orc.OrcEncoding.DWRF;
+import static com.facebook.presto.orc.OrcTester.HIVE_STORAGE_TIME_ZONE;
+import static com.facebook.presto.orc.OrcTester.mapType;
+import static com.facebook.presto.orc.OrcWriteValidation.OrcWriteValidationMode.BOTH;
+import static com.facebook.presto.orc.metadata.CompressionKind.ZSTD;
+import static com.google.common.io.Files.createTempDir;
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static java.util.UUID.randomUUID;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+@State(Scope.Thread)
+@OutputTimeUnit(MILLISECONDS)
+@Fork(3)
+@Warmup(iterations = 10, time = 1000, timeUnit = MILLISECONDS)
+@Measurement(iterations = 20, time = 1000, timeUnit = MILLISECONDS)
+@BenchmarkMode(Mode.AverageTime)
+public class BenchmarkMapFlatWriter
+{
+    private static final int SEED = 0;
+
+    public static void main(String[] args)
+            throws Throwable
+    {
+        Options options = new OptionsBuilder()
+                .verbosity(VerboseMode.NORMAL)
+                .include(BenchmarkMapFlatWriter.class.getName() + ".*")
+                .build();
+
+        new Runner(options).run();
+    }
+
+    @Benchmark
+    public void baseline(BenchmarkData data)
+            throws IOException
+    {
+        doWrite(data);
+    }
+
+    private void doWrite(BenchmarkData data)
+            throws IOException
+    {
+        OrcWriterOptions orcWriterOptions = OrcWriterOptions.builder()
+                .withFlattenedColumns(ImmutableSet.of(0))
+                .build();
+
+        OrcWriter writer = new OrcWriter(
+                new OutputStreamDataSink(new FileOutputStream(data.orcFile)),
+                ImmutableList.of("col1"),
+                ImmutableList.of(data.type),
+                DWRF,
+                ZSTD,
+                Optional.empty(),
+                NO_ENCRYPTION,
+                orcWriterOptions,
+                ImmutableMap.of(),
+                HIVE_STORAGE_TIME_ZONE,
+                false,
+                BOTH,
+                new NoOpOrcWriterStats());
+
+        for (Block block : data.blocks) {
+            writer.write(new Page(block));
+        }
+
+        writer.close();
+    }
+
+    @State(Scope.Thread)
+    public static class BenchmarkData
+    {
+        private final Random random = new Random(SEED);
+
+        private Type type;
+        private File temporaryDirectory;
+        private File orcFile;
+
+        @Param("bigint")
+        private String keyType = "bigint";
```

Comment:
These fields have default values to be able to run the benchmark methods outside of the JMH.